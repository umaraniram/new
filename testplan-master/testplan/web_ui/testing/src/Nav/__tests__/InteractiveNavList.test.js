/* Unit tests for the InteractiveNavList component. */
import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import InteractiveNavList from '../InteractiveNavList.js';
import {FakeInteractiveReport} from '../../Common/sampleReports.js';

describe('InteractiveNavList', () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders and matches snapshot', () => {
    const renderedNavList = shallow(
      <InteractiveNavList
        entries={FakeInteractiveReport.entries}
        breadcrumbLength={1}
        width={'28em'}
        handleNavClick={() => undefined}
        autoSelect={() => undefined}
        filter={null}
        displayEmpty={true}
        displayTags={false}
        displayTime={false}
        handlePlayClick={(e) => undefined}
        envCtrlCallback={(e, action) => undefined}
      />
    );
    expect(renderedNavList).toMatchSnapshot();
  });
});

