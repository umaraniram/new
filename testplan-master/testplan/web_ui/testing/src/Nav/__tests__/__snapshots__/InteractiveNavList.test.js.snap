// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractiveNavList shallow renders and matches snapshot 1`] = `
<Column
  width="28em"
>
  <ListGroup
    className="buttonList_mjw45f"
    tag="ul"
  >
    <ListGroupItem
      className="navButton_1x055n9-o_O-navButtonInteract_1ofjo98-o_O-unselectable_16maxrd"
      key="bbb"
      onClick={[Function]}
      tabIndex="1"
      tag="li"
    >
      <InteractiveNavEntry
        caseCountFailed={0}
        caseCountPassed={0}
        envCtrlCallback={[Function]}
        handlePlayClick={[Function]}
        name="Interactive MTest"
        runtime_status="ready"
        status="unknown"
        type="multitest"
      />
    </ListGroupItem>
  </ListGroup>
</Column>
`;
